plugins/statuslines/lualine.nix

```diff
@@ -1,260 +1,281 @@
-              default = { };
-              description = "extra options for the component";
+              })) 1 "Adds padding to the left and right of components.";
+
+              fmt = lib.nixvim.mkNullOrLuaFn' {
+                description = ''
+                  A lua function to format the component string.
+                '';
+                example = ''
+                  ```lua
+                    function(text)
+                      return text .. "!!!"
+                    end
+                  ```
+                '';
+              };
             };
           };
-        })
-      )
-    ) "";
+        in
+        listOf (lib.nixvim.transitionType oldType coerceFn newType);
+      description = "";
+    };
 
   mkEmptySectionOption = name: {
-    lualine_a = mkComponentOptions "";
-    lualine_b = mkComponentOptions "";
-    lualine_c = mkComponentOptions "";
-    lualine_x = mkComponentOptions "";
-    lualine_y = mkComponentOptions "";
-    lualine_z = mkComponentOptions "";
+    lualine_a = mkComponentOptions "Left section on left side.";
+    lualine_b = mkComponentOptions "Middle section on left side.";
+    lualine_c = mkComponentOptions "Right section on left side.";
+    lualine_x = mkComponentOptions "Left section on right side.";
+    lualine_y = mkComponentOptions "Middle section on right side.";
+    lualine_z = mkComponentOptions "Right section on right side.";
   };
 in
-{
-  options = {
-    plugins.lualine = {
-      enable = mkEnableOption "lualine";
-
-      package = lib.mkPackageOption pkgs "lualine" {
-        default = [
-          "vimPlugins"
-          "lualine-nvim"
-        ];
-      };
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "lualine";
+  originalName = "lualine-nvim";
+
+  package = lib.mkPackageOption pkgs "lualine" {
+    default = [
+      "vimPlugins"
+      "lualine-nvim"
+    ];
+  };
 
-      gitPackage = lib.mkPackageOption pkgs "git" {
-        nullable = true;
-      };
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  # TODO: Added 2024-09-05, remove after 24.11
+  optionsRenamedToSettings = [
+    "extensions"
+    "sections"
+    "inactiveSections"
+    "tabline"
+    "winbar"
+    "inactiveWinbar"
+  ];
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "lualine"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+      optionsPath = settingsPath ++ [ "options" ];
+      oldOptions = [
+        "theme"
+        "globalstatus"
+        "refresh"
+        "iconsEnabled"
+        "sectionSeparators"
+        "componentSeparators"
+        "disabledFiletypes"
+        "ignoreFocus"
+        "alwaysDivideMiddle"
+      ];
+    in
+    mkSettingsRenamedOptionModules basePluginPath optionsPath oldOptions;
 
-      iconsEnabled = mkOption {
-        type = types.bool;
-        description = "Whether to enable/disable icons for all components.";
-        default = true;
-      };
+  settingsOptions = {
+    options = {
```

Comment:
Typo: 
```suggestion
  settingsOptions = {
```

Reply:
Ah, my mistake: https://github.com/nix-community/nixvim/blob/7409e80bd2dbbc5d653b27170d2fb33082150df3/plugins/statuslines/lualine.nix#L236-L237

Reply:
That's not a typo, the plugin has a table of options.

Reply:
https://github.com/nvim-lualine/lualine.nvim/blob/b431d228b7bbcdaea818bdc3e25b8cdbe861f056/lua/lualine/config.lua#L10